starlark/library.go

```diff
@@ -69,12 +69,41 @@ func init() {
 	}
 }
 
-type builtinMethod func(b *Builtin, args Tuple, kwargs []Tuple) (Value, error)
+type builtinMethod func(thread *Thread, b *Builtin, args Tuple, kwargs []Tuple) (Value, error)
+
+type builtins struct {
+	attrs map[string]*Builtin
+	names []string
+}
+
+func (bs *builtins) bindAttr(recv Value, name string) (Value, error) {
```

Comment:
Once you've made builtins just a map, these two methods can go back to being standalone functions.